cmd/notification/main.go

```diff
@@ -0,0 +1,184 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/argoproj/notifications-engine/pkg/api"
+	"github.com/argoproj/notifications-engine/pkg/controller"
+	"github.com/argoproj/notifications-engine/pkg/services"
+	"github.com/argoproj/notifications-engine/pkg/subscriptions"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"k8s.io/klog/v2/klogr"
+	csclientv1beta1 "kpt.dev/configsync/clientgen/apis/typed/configsync/v1beta1"
+	"kpt.dev/configsync/pkg/api/configsync/v1beta1"
+	"kpt.dev/configsync/pkg/core"
+	"kpt.dev/configsync/pkg/kinds"
+	"kpt.dev/configsync/pkg/util"
+	"kpt.dev/configsync/pkg/util/log"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/cache"
+)
+
+const objectKey = "sync"
+
+var (
+	apiGroup = flag.String("api-group", util.EnvString(util.NotificationAPIGroup, v1beta1.SchemeGroupVersion.Group),
+		"Group of the Resource for notifications")
+	apiVersion = flag.String("api-version", util.EnvString(util.NotificationAPIVersion, v1beta1.SchemeGroupVersion.Version),
+		"Version of the Resource for notifications")
+	apiKind = flag.String("api-kind", os.Getenv(util.NotificationAPIKind),
+		"Resource information of the Resource for notifications")
+	resourceName = flag.String("resource-name", os.Getenv(util.NotificationResourceName),
+		"Name of the Resource to be notified")
+	resourceNamespace = flag.String("resource-namespace", os.Getenv(util.NotificationResourceNamespace),
+		"Namespace of the Resource to be notified")
+	resyncCheckPeriod = flag.Duration("resync-period", pollingPeriod(util.NotificationResyncPeriod, time.Minute),
+		"Period of time between checking if the apiResource listener needs a resync")
+	cmName = flag.String("cm-name", os.Getenv(util.NotificationConfigMapName),
+		"Name of the ConfigMap for the notification configs")
+	secretName = flag.String("secret-name", os.Getenv(util.NotificationSecretName),
+		"Name of the Secret for the notification configs")
+)
+
+// pollingPeriod parses the polling duration from the environment variable.
```

Comment:
Duplicate with https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/reconcilermanager/controllers/util.go#L296.

Reply:
Good catch, updated to use the shared util